nixos/modules/hardware/video/nvidia.nix

```diff
@@ -343,14 +343,20 @@ in
             ];
 
             # Don't add `nvidia-uvm` to `kernelModules`, because we want
-            # `nvidia-uvm` be loaded only after `udev` rules for `nvidia` kernel
-            # module are applied.
+            # `nvidia-uvm` be loaded only after the GPU device is available, i.e. after `udev` rules
+            # for `nvidia` kernel module are applied.
+            # This matters on Azure GPU instances: https://github.com/NixOS/nixpkgs/pull/267335
             #
             # Instead, we use `softdep` to lazily load `nvidia-uvm` kernel module
             # after `nvidia` kernel module is loaded and `udev` rules are applied.
             extraModprobeConfig = ''
               softdep nvidia post: nvidia-uvm
             '';
+
+            # Exception is the open-source kernel module failing to load nvidia-uvm using softdep,
+            # for unknown reasons.
+            # This breaks CUDA: https://github.com/NixOS/nixpkgs/issues/334180
```

Comment:
We can remove the [part in the X11 section](https://github.com/NixOS/nixpkgs/blob/5fa505109481da78b5a3555b69e4168faf9c112f/nixos/modules/hardware/video/nvidia.nix#L648-L651), since we're setting it in the common section here, which will also apply to all other sections.

Reply:
Damn, I this was already handled before! https://github.com/NixOS/nixpkgs/pull/334340